src/components/ConfigEditor/ConnectionConfig.tsx

```diff
@@ -29,7 +29,10 @@ const ConnectionConfig: React.FC<ConnectionConfigProps> = ({
     if (!jsonData.azureCloud) {
       updateJsonData('azureCloud', AzureCloudType.AzurePublic);
     }
-  }, [jsonData.azureCloud, updateJsonData]);
+    if (!!jsonData.oauthPassThru !== !!jsonData.onBehalfOf) {
+      updateJsonData('oauthPassThru', !!jsonData.onBehalfOf);
+    }
+  }, [jsonData.azureCloud, jsonData.oauthPassThru, jsonData.onBehalfOf, updateJsonData]);
```

Comment:
I think it's simpler to just set `oauthPassThru` in the function of the line 174, isn't it?

Reply:
UseEffect is also called on the initial value, which applies the parameter binding to the (provisioning) configuration as well. I know little about React so feel free to correct me there.

Reply:
If you provision ADX, this will only work if you actually go in and open the configuration page, and that is not ideal. Instead of setting the `oauthPassThru` flag on mount of this page, it's better to set this when changing the `onBehalfOf` switch (as Andrés was suggesting). You can also add a new section in the readme with examples of how to provision ADX with and without OBO. 

Example without: 
```yaml
  - name: Azure Data Explorer
    type: grafana-azure-data-explorer-datasource
    access: proxy
    basicAuth: false
    editable: true
    jsonData:
      clientId: '<your client id>'
      clusterUrl: '<your cluster url>'
      tenantId: '<your tenant id>'
      tlsAuth: false
      tlsAuthWithCACert: false
      defaultDatabase: '<your deault database>'
    secureJsonData:
      clientSecret: '<your client secret>'
      tlsCACert: ''
      tlsClientCert: ''
      tlsClientKey: ''
```

With OBO enabled: 
```yaml
  - name: Azure Data Explorer
    type: grafana-azure-data-explorer-datasource
    access: proxy
    basicAuth: false
    editable: true
    jsonData:
      oauthPassThru: true
      onBehalfOf: true
      clientId: '<your client id>'
      clusterUrl: '<your cluster url>'
      tenantId: '<your tenant id>'
      tlsAuth: false
      tlsAuthWithCACert: false
      defaultDatabase: '<your deault database>'
    secureJsonData:
      clientSecret: '<your client secret>'
      tlsCACert: ''
      tlsClientCert: ''
      tlsClientKey: ''
```

See [cloudwatch docs](https://github.com/grafana/grafana/blob/main/docs/sources/datasources/aws-cloudwatch/provision-cloudwatch.md#L18) for inspiration

Reply:
I believe that is exactly what happens now @sunker. No need to specify `oauthPassThru`—it will follow whatever the `onBehalfOf` is.

Reply:
Yeah but only if you actually open the configuration page? That's when the effect is triggered. When you provision the app, you don't want to go into the config page - you want to be able to go straight to the dashboard and have a working config.

Reply:
I believe provisioning works without opening the configuration page now. Please let me know if it doesn't.

Reply:
OK, a duplicate provisioning file prevented me from seeing this. Yeah, one way would be to document the workaround.
Does the React setup really prevent us from fixing the issue? Where does the plugin receive the configuration?

Reply:
[This](https://github.com/grafana/azure-data-explorer-datasource/blob/91fd99784426b32e490ba8bb1dec3fc5e3808947/pkg/azuredx/datasource.go#L33) is where the plugin receives new settings. In this method you could check if user has bad conigurations, i.e if `onBehalfOf` whereas `oauthPassThru` is missing (or set to false). In that case, you can log an error. 

Please add a new section with the header `Configure the data source with provisioning` at the bottom of the readme file. Add the two examples from above. You can use the documentation from [timestream data source](https://github.com/grafana/timestream-datasource/blob/main/src/README.md#configure-the-data-source-with-provisioning) as inspiration when formatting the markdown.

Reply:
Setting `oauthPassThru` in the Go code won't affect the configuration unfortunately. The user should not be bothered by such internal detail. There must be a place in this JavaScript mess where the configuration is read and applied. Is it still reachable, or are we lost in layers of abstraction?

Reply:
>Setting oauthPassThru in the Go code won't affect the configuration unfortunately.

I know it won't. We can't prevent user's from adding bad configurations while using provisioning, but we can print a log message to let them know that the config was bad. This is mainly for the provisioning story. 

If the user configures the data source through the UI (not provisioning), we can prevent them from having bad configuration using code. You should make sure to sync the `oauthPassThru` with the `onBehalfOf` prop on line 188 in the ConnectionConfig.tsx.